mleap-runtime/src/main/scala/ml/combust/mleap/runtime/transformer/Transformer.scala

```diff
@@ -33,6 +34,8 @@ trait Transformer extends AutoCloseable {
     */
   def transform[TB <: TransformBuilder[TB]](builder: TB): Try[TB]
 
+  def getSchema(): Try[Seq[StructField]]
```

Comment:
Oh I see, this is the field generating method. Maybe we can call this "getFields" instead and then make another method that uses it to return a Try[StructType] called "getSchema"